### ISSUE TITLE
[NEW DEVICE] - MyQ - Gate

### ISSUE THREAD

**Issue by jlinscott** [pyalarmdotcom/alarmdotcom#178]
**Which device would you like us to support?**
I have my MyQ enabled garage doors and driveway gate linked to my alarm.com account. It works well on the alarm.com app and on the web portal, but the gate does not link to HA. The four garage doors do, but the gate does not. I would like the gate to be a device on HA. It's odd that the integration into alarm.com from MyQ only integrates the four 

**Where does this device appear in the Alarm.com mobile app?**
It shows up in a separate "card" titled GATES

**What types of actions does this device support?**
Opening and closing

**Which provider do you use?**
Custom Automated Technologies / alarm.com

**Include pyalarmdotcomajax Server Output**
I am not sure how to do this without possibly compromising my login credentials.

**Include Action Endpoints**
I am not sure how to do this but presume it's nearly identical to garage doors.

See screenshot attached.
![Screenshot_20221110-001041](https://user-images.githubusercontent.com/12158739/201006300-40ffaddd-86e7-4502-8cd7-becf474d9ba0.png)

**Comment by jlinscott** [comment]
@elahd (or anyone else) - is there anything I can provide to help someone look into the above?

**Comment by elahd** [comment]
Can you run through a few command line steps to share the data structure for gates?

1. Install [Python >= 3.9](https://www.python.org/downloads/).
2. Install [pyalarmdotcomajax](https://github.com/pyalarmdotcom/pyalarmdotcomajax) via pip: `pip install pyalarmdotcomajax`.
3. Run `adc -v -u "your_username" -p "your_password" get -x`
4. Paste all of the text that appears in the `====[ Gate ]====` section.

Just a heads up that the pasted data _may_ leak sensitive information. (I.e.: We'll see what you named your garage door.) Give the text a quick scan and redact anything that you don't want posted.

Thanks!

**Comment by jlinscott** [comment]
@elahd Thank you!  See below:

←[1m←[43m←[30m====[ Gate ]====←[0m

{"data": [{"id": "103168694-2209", "type": "devices/gate", "attributes": {"state": 2, "desiredState": 2, "supportsRemoteClose": true, "managedDeviceType": 5, "hasState": true, "canBeRenamed": true, "canBeDeleted": false, "canAccessWebSettings": true, "canAccessAppSettings": true, "webSettings": 1004, "canAccessTroubleshootingWizard": false, "troubleshootingWizard": null, "addDeviceResource": 0, "canBeAssociatedToVideoDevice": false, "associatedCameraDeviceIds": {}, "macAddress": "", "manufacturer": "Chamberlain", "isAssignedToCareReceiver": false, "isOAuth": true, "isZWave": false, "supportsCommandClassBasic": false, "isMalfunctioning": false, "canBeSaved": true, "canChangeDescription": true, "description": "Driveway Gate", "deviceModelId": 9475, "canConfirmStateChange": true, "canReceiveCommands": true, "remoteCommandsEnabled": true, "hasPermissionToChangeState": true, "deviceIcon": {"icon": 305}, "batteryLevelNull": null, "lowBattery": false, "criticalBattery": false}, "relationships": {"system": {"data": {"id": "13207056", "type": "systems/system"}}, "stateInfo": {"data": {"id": "103168694-2209-2", "type": "devices/state-info"}}}}], "included": [], "meta": {"transformer_version": "1.1"}}

**Comment by elahd** [comment]
Thanks. Try this:

1. In Home Assistant, go to HACS and click on the Alarm.com integration.
2. Click on the three dots on the top right, then click on "Redownload".
3. From the "Select version" menu, select "master", then click download.
4. Restart Home Assistant.

Does your gate show up? Can you open and close it?

**Comment by jlinscott** [comment]
Thank you very much @elahd 
I was able to add it to Home Assistant, but it is not functional. It does not allow me to open or close the gate.  Please let me know what to try next.

**Comment by elahd** [comment]
It would help if you're able to share the debug logs that are generated after you try to open or close the gate. I'm coding in the dark as I don't have a gate myself! You can enable debugging by adding the following to your configuration.yaml, followed by rebooting:

```yaml
logger:
  default: warn
  logs:
    custom_components.alarmdotcom: debug
    pyalarmdotcomajax: debug
```

Thanks!

**Comment by jlinscott** [comment]
Thank you @elahd 

I cannot click to open or close the gate. It is almost like it is greyed-out. See image below and compare against the garage doors, which I can currently (because they are now closed) open (they are in white).
![image](https://user-images.githubusercontent.com/12158739/208152308-dbac0537-ca8e-483d-a809-16154471cb52.png)

**Comment by elahd** [comment]
Do you see anything relavent in the logs? 

Also, can you press the debug button on the gate in Home Assistant and paste the debug logs here?

First:

![image](https://user-images.githubusercontent.com/466460/208168317-d224e870-612b-4f04-a588-a4cb7dba09b2.png)

Then check HA's logs for a debug entry:

![image](https://user-images.githubusercontent.com/466460/208168337-2de8d685-b2ab-4927-b81c-e9ffbb3a6fb0.png)

![image](https://user-images.githubusercontent.com/466460/208168349-e5e1790c-8064-4636-98ee-78141cafdce2.png)

**Comment by jlinscott** [comment]
This is all I see, @elahd 

![image](https://user-images.githubusercontent.com/12158739/208179191-597b15f8-ce8e-4b50-af0a-02db7924993f.png)

**Comment by jlinscott** [comment]
@elahd 
I found out that when the gate is open, it shows the "close" function and it works perfectly.  It seems that it does not recognize when the gate is shut and it can be opened.
Not sure if this helps, but thought I'd share.

**Comment by jlinscott** [comment]
Hi @elahd any update? Thank you!

**Comment by elahd** [comment]
Hey -- sorry for the radio silence! A bit tied up with the holidays but I'll get back to this soon.

**Comment by jlinscott** [comment]
@elahd Thank you and happy holidays!

**Comment by elahd** [comment]
Thanks!

Try again now. Redownload from the master branch and restart HA.

**Comment by jlinscott** [comment]
Hi @elahd I tried this and it does not work. The previous function of closing it when it was left open does still work, though.

**Comment by elahd** [comment]
Is the open button still disabled? Also, do you see any errors in Home Assistant's logs?

**Comment by elahd** [comment]
Actually, forget that. It looks like my fix didn't make it into the master branch. It's there now. Can you redownload/reset and test once more?

**Comment by jlinscott** [comment]
@elahd It works! Thank you!!

**Comment by elahd** [comment]
Awesome! It's now released as v2.0.19. Thanks for testing.

**Comment by jlinscott** [comment]
Thanks again!
